src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -5335,100 +5335,114 @@ and CheckForAdjacentListExpression (cenv: cenv) synExpr hpa isInfix delayed (arg
 /// method applications and other item-based syntax.
 and TcExprThen (cenv: cenv) overallTy env tpenv isArg synExpr delayed =
     let g = cenv.g
+    
+    let cachedExpression = 
+        match env.eCachedImplicitYieldExpressions.TryFind synExpr.Range with
+        | Some (se, ty, e) ->
```

Comment:
I suggested the multimap for the case of multiple nodes sharing the same range - but looking at parser code, it might be non trivial to come up with code that uses an implicit yield and then a different synexpr which happens to hold the same range.


My thinking was:
- from the multimap, get the N entries for the same range (with N typically being 0 or 1)
- pick the right value using reference equality on `synExpr` from this small list